recipe/meta.yaml

```diff
@@ -11,40 +11,41 @@ source:
 
 build:
   script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
-  number: 1
-  skip: true  # [win]
+  number: 0
+  # pykrb5 isn't supported on win64, it fails because of an error:
+  # '"krb5-config --cflags krb5" is not recognized as an internal or external command'
+  skip: true  # [py<37 or win]
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - cython                                 # [build_platform != target_platform]
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - {{ compiler('c') }}
   host:
-    - krb5
-    - python
     - cython >=0.29.32,<4.0.0
-    - setuptools >=42.0.0
+    - krb5 1.20
     - pip
+    - python
+    - setuptools
+    - wheel
   run:
-    - cryptography >=1.3
-    - krb5
+    - krb5 >=1.20.1,<1.21.0a0
```

Comment:
The README says that `heimdal` will be used on MacOS from the source folder `build_helpers/heimdal` :
```
KRB5_MACOS_HEIMDAL_DIR
Used when compiling on macOS to point to the Heimdal install directory
Used to find the Heimdal header files as macOS does not include, or provide a way to obtain, these header files for their Heimdal framework
Defaults to {git_root}/build_helpers/heimdal 
```

Reply:
`conda search heimdal` says that we do not have `heimdal`.

Reply:
@skupr-anaconda Please take a look at the links I provided. Also look at https://github.com/jborean93/pykrb5/blob/v0.5.1/setup.py#L182.

Reply:
If I run tests with `k5test` then a lot of tests fail with the error `FileNotFoundError: [Errno 2] No such file or directory: '/usr/sbin/kdb5_util'`. That's weird as `kdb5_util` is located in `$PREFIX/sbin` and `not` in `/usr/sbin`

Reply:
Ah, `k5test` incorrectly defines the place of `kdb5_util`:
```
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/k5test/realm.py:517: in create_kdb
    self.run([self.kdb5_util, "create", "-W", "-s", "-P", "master"])
```
see also https://github.com/pythongssapi/k5test/blob/17512bd2137ffcdd1fdb7ff4210891ad34d27803/k5test/realm.py#L459

In this case, the tests are useless

Reply:
I tried to set `KRB5_MAIN_LIB` but then `import krb5` in the test environment failed:
`symbol not found in flat namespace '_krb5_data_free'`

Reply:
@skupr-anaconda you can apply this diff to use our krb5 instead of using the Private macOS Heimdal framework:
```diff
diff --git a/setup.py b/setup.py
index 1d27cf9..66e4d9e 100755
--- a/setup.py
+++ b/setup.py
@@ -125,10 +125,12 @@ def get_krb5_lib_path(
     elif not krb5_lib:
         for opt in libraries:
             if opt.startswith("krb5"):
-                ext = {
-                    "nt": "dll",
-                    "darwin": "dylib",
-                }.get(os.name, "so")
+                if os.name == "nt":
+                    ext = "dll"
+                elif sys.platform == "darwin":
+                    ext = "dylib"
+                else:
+                    ext = "so"
 
                 krb5_lib = f"lib{opt}.{ext}"
                 break
@@ -159,9 +161,6 @@ if not SKIP_EXTENSIONS:
     print(f"Using krb5-config at '{kc}'")
 
     macos_native = False
-    if sys.platform == "darwin":
-        mac_ver = [int(v) for v in platform.mac_ver()[0].split(".")]
-        macos_native = mac_ver >= [10, 7, 0]
 
     compile_args, raw_link_args = [
         shlex.split(os.environ[e], posix=True) if e in os.environ else None
```

Reply:
Let me try...

Reply:
With this patch, logs messages are:
```
...
  Skipping src/krb5/_creds_opt_heimdal.pyx as it is not supported by the selected Kerberos implementation.
  Compiling src/krb5/_creds_opt_mit.pyx
  Compiling src/krb5/_creds_opt_set_in_ccache.pyx
  Compiling src/krb5/_creds_opt_set_pac_request.pyx
  Compiling src/krb5/_exceptions.pyx
  Compiling src/krb5/_keyblock.pyx
  Compiling src/krb5/_kt.pyx
  Compiling src/krb5/_kt_mit.pyx
  Skipping src/krb5/_kt_heimdal.pyx as it is not supported by the selected Kerberos implementation.
  Compiling src/krb5/_kt_have_content.pyx
  Compiling src/krb5/_principal.pyx
  Skipping src/krb5/_principal_heimdal.pyx as it is not supported by the selected Kerberos implementation.
...
  Skipping src/krb5/_creds_opt_heimdal.pyx as it is not supported by the selected Kerberos implementation.
  Compiling src/krb5/_creds_opt_mit.pyx
  Compiling src/krb5/_creds_opt_set_in_ccache.pyx
  Compiling src/krb5/_creds_opt_set_pac_request.pyx
  Compiling src/krb5/_exceptions.pyx
  Compiling src/krb5/_keyblock.pyx
  Compiling src/krb5/_kt.pyx
  Compiling src/krb5/_kt_mit.pyx
  Skipping src/krb5/_kt_heimdal.pyx as it is not supported by the selected Kerberos implementation.
  Compiling src/krb5/_kt_have_content.pyx
  Compiling src/krb5/_principal.pyx
  Skipping src/krb5/_principal_heimdal.pyx as it is not supported by the selected Kerberos implementation.



```
and 1 new test fails:
```
    @pytest.mark.requires_api("cc_support_switch")
    def test_cc_supports_switch(cache_type: bytes, expected: bool) -> None:
        actual = krb5.cc_support_switch(krb5.init_context(), cache_type)
>       assert actual is expected
E       assert True is False

tests/test_ccache.py:230: AssertionError
```

Reply:
Yep, you can patch the test or skip it. That's basically just a difference between the real krb5 and macOS's private krb5 framework, at least from what I understand. See the comment at https://github.com/jborean93/pykrb5/blob/432cc45a0705cf84caa5fc5c94ac535f9c85a6cc/tests/test_ccache.py#L222.

Reply:
Skipping it with a comment

Reply:
Well, `osx-64` crashed:
`    subprocess.CalledProcessError: Command '['/var/folders/c_/qfmhj66j0tn016nkx_th4hxm0000gp/T/abs_1b62n2oerf/croot/pykrb5_1714056117212/_build_env/bin/llvm-otool', '-l', '/var/folders/c_/qfmhj66j0tn016nkx_th4hxm0000gp/T/abs_1b62n2oerf/croot/pykrb5_1714056117212/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/site-packages/krb5/_kt_have_content.cpython-38-darwin.so']' died with <Signals.SIGABRT: 6>.`

Reply:
I restarted the failed build and it passed. I'm not too worried with this failure. `llvm-otool` crashed, which is not a big deal.